src/components/CIPipelineN/EnvironmentList.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { createClusterEnvGroup } from '../common'
+import ReactSelect from 'react-select'
+import { Environment } from '../cdPipeline/cdPipeline.types'
+import { groupStyle } from '../secrets/secret.utils'
+import { GroupHeading } from '../v2/common/ReactSelect.utils'
+import { DropdownIndicator } from '../cdPipeline/cdpipeline.util'
+
+export function EnvironmentList({ isBuildStage, environments, selectedEnv, setSelectedEnv }:
+    { isBuildStage: boolean, environments: any[], selectedEnv: Environment, setSelectedEnv?: (_selectedEnv: Environment) => void | React.Dispatch<React.SetStateAction<Environment>> }) {
+
+    const selectEnvironment = (selection: Environment) => {
+        const _selectedEnv = environments.find((env) => env.id == selection.id)
+        setSelectedEnv(_selectedEnv)
+    }
+
+
+    const envList = createClusterEnvGroup(environments, 'clusterName')
+
+    const groupHeading = (props) => {
```

Comment:
Please move these const to constants file

Reply:
done